src/kernel/arch/x86/paging.zig

```diff
@@ -142,8 +144,14 @@ pub fn init(mem_profile: *const MemProfile, allocator: *std.mem.Allocator) void
 
     mapDir(kernel_directory, p_start, p_end, v_start, v_end, allocator) catch unreachable;
     const dir_physaddr = @ptrToInt(kernel_directory) - @ptrToInt(&KERNEL_ADDR_OFFSET);
-    asm volatile ("mov %[addr], %%cr3" :: [addr] "{eax}" (dir_physaddr));
-    isr.registerIsr(14, pageFault) catch unreachable;
+    asm volatile ("mov %[addr], %%cr3"
```

Comment:
Idk if I mentioned this before, but could this be moved to the arch.zig for x86. This will then allow you to unit test the init function by mocking out all the arch specific calls.

Reply:
The only things in the arch.zig file should be things that all architectures can do and paging may not be one of them. Why do I need to do mocking for the init function? Runtime tests catch all the testable functionality anyway.

Reply:
I mean the assembly, not the whole init function lol. As I have the assembly of loading the GDT and IDT there. Then can write a unit test that will mock out the assembly and check the addresse being passed to the cr3 register.

Reply:
Yeah I mean  the assembly too :p I'm not even sure if lgdt and lidt should be in there either, since they're only used by one architecture and one file each.

Reply:
Hmmm, but I think that all the x86 files will be cluttered with assembly and having it all in one place makes the code base neater. Maybe could have a assembly file instead of putting it in the arch.zig

Reply:
Not sure I agree, I think adding more files makes it less neat. Really, inline assembly is no different to normal code so I don't see why it all has to be moved to one place.

Reply:
ok, you win for now :P. You may keep the assembly here